src/frontend/js/data/SessionProvider/index.spec.tsx

```diff
@@ -0,0 +1,193 @@
+  });
+  it('uses JoanieSessionProvider if joanie is enabled', async () => {
+    jest.doMock('../../utils/api/joanie.ts', () => ({
+      isJoanieEnabled: true,
+    }));
+    jest.doMock('./BaseSessionProvider', () => ({
+      __esModule: true,
+      default: () => 'BaseSessionProvider',
+    }));
+    jest.doMock('./JoanieSessionProvider', () => ({
+      __esModule: true,
+      default: () => 'JoanieSessionProvider',
+    }));
+    const { SessionProvider: Provider } = await import('.');
+
+    render(<Provider />);
+
+    await screen.findByText('JoanieSessionProvider');
+  });
+
+  // - useSession Provider test suite
+  describe('useSession', () => {
+    it('provides a null user if whoami return 401', async () => {
+      const userDeferred = new Deferred();
+      fetchMock.get('https://endpoint.test/api/user/v1/me', userDeferred.promise);
+
+      const { result } = renderHook(() => useSession(), {
+        wrapper,
+        initialProps: { client: queryClient },
+      });
+
+      await act(async () => userDeferred.resolve(401));
+
+      expect(result.current.user).toBeNull();
+      expect(result.all).toHaveLength(2);
+
+      await act(async () => {
+        jest.runOnlyPendingTimers();
+      });
+
+      const cacheString = sessionStorage.getItem(REACT_QUERY_SETTINGS.cacheStorage.key);
+      expect(cacheString).not.toBeNull();
+
+      const client = JSON.parse(cacheString!);
+      expect(client.clientState.queries).toHaveLength(1);
+      expect(client.clientState.queries[0].queryKey).toEqual('user');
+    });
+
+    it('provides user infos if user is authenticated then stores in cache', async () => {
+      const username = faker.internet.userName();
+      const userDeferred = new Deferred();
+      fetchMock.get('https://endpoint.test/api/user/v1/me', userDeferred.promise);
+      const { result } = renderHook(useSession, { wrapper });
+
+      await act(async () => {
+        userDeferred.resolve({ username });
+        jest.runAllTimers();
+      });
+
+      expect(result.current.user).toStrictEqual({ username });
+      expect(result.all).toHaveLength(2);
+      const cacheString = sessionStorage.getItem(REACT_QUERY_SETTINGS.cacheStorage.key);
+      expect(cacheString).not.toBeNull();
+      expect(cacheString).toContain(username);
+    });
+
+    it('destroy session then logout', async () => {
+      const username = faker.internet.userName();
+      const userDeferred = new Deferred();
+
+      fetchMock.get('https://endpoint.test/api/user/v1/me', userDeferred.promise);
+      fetchMock.get('https://endpoint.test/logout', 200);
+      const { result } = renderHook(useSession, { wrapper });
+
+      await act(async () => {
+        userDeferred.resolve({ username });
+        jest.runAllTimers();
+      });
+
+      const { user, destroy } = result.current;
+      expect(user).toStrictEqual({ username });
+      expect(sessionStorage.getItem(REACT_QUERY_SETTINGS.cacheStorage.key) || '').toContain(
```

Comment:
Dose using `|| ''` provdes better information when it fails ?

Reply:
Nope I could remove this fallback.

In fact when we use fallback we get this error message when the test fails : 
```
Error: expect(received).toMatch(expected)

Expected pattern: /"data":null,.*"queryKey":"user"/
Received string:  ""
```

Otherwise we get :
```
Error: expect(received).toMatch(expected)

Matcher error: received value must be a string

Received has value: null
```